src/libraries/Native/Unix/System.Native/pal_memory.c

```diff
@@ -2,9 +2,26 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 #include "pal_memory.h"
+
+#include <stdlib.h>
 #include <string.h>
 
-void* SystemNative_MemSet(void *s, int c, uintptr_t n)
+void* SystemNative_MemAlloc(uintptr_t size)
+{
+    return malloc(size);
```

Comment:
There is a potential platform inconsistency. The spec for `malloc` leaves the return value if `size` == 0 up to the platform. When I have written PALs in the past, I typically make this explicit and if `size` == 0, return `nullptr` OR if `size` == 0 increase it to 1 so we get an allocation.

See https://en.cppreference.com/w/c/memory/malloc for details. Key phrase:

>If size is zero, the behavior of malloc is implementation-defined.

Reply:
Looks like this is being handled in the managed layer -  yay!

Reply:
I put it the `+1` into the managed. It seems to be a tiny bit more efficient when it is part of the bigger function, and this PAL layer is a pure forwarder.

Reply:
Just curious why is there value in allocation if they requested 0?

Reply:
Some platforms return non-null values when `0` is supplied to `malloc()`. It is considered an implementation detail and some return non-null some return null so we need to be consistent and the best way to do that is if `0` is supplied up to `1` so a non-null is always returned - otherwise it is out of memory.